lib/include/ert/enkf/row_scaling.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+  Copyright (C) 2020  Equinor ASA, Norway.
+
+  The file 'row_scaling.hpp' is part of ERT - Ensemble based Reservoir Tool.
+
+  ERT is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+  FITNESS FOR A PARTICULAR PURPOSE.
+
+  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+  for more details.
+*/
+
+#ifndef ROW_SCALING_H
+#define ROW_SCALING_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <ert/util/type_macros.h>
+#include <ert/res_util/matrix.hpp>
+
+typedef struct row_scaling_struct row_scaling_type;
+
+row_scaling_type * row_scaling_alloc();
+row_scaling_type * row_scaling_alloc_copy(const row_scaling_type * row_scaling);
+void               row_scaling_free(row_scaling_type * row_scaling);
+void               row_scaling_free__(void * arg);
```

Comment:
Private function in public interface.

Also is it used anywhere?

Reply:
Yes it is used - when a `row_scaling` instance is registered in the `local_dataset::` - might still be on my machine though; this is certainly WIP.